### PR TITLE
[chore] Fix issues with loading User Account Extension

### PR DESCRIPTION
**Pull request by theBeginner86**
**Notes for Reviewers**

This PR addresses issues with loading user account extension in Meshery UI.

**Following issues are resolved:**
- [x] Multiple reloading
- [x] Multiple fetches to determine capabilities 
- [x] Redirection not working in few scenarios for playground 

This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Curious: what is `type` is used for?

**Review by leecalcote — CHANGES_REQUESTED** [review]
Is there an update to the https://docs.meshery.io/extensibility/providers#ui-extension-points to accompany this?

**Comment by theBeginner86** [comment]
> Curious: what is `type` is used for?

`type` means the type of remote extension.
Could be:
1. Navigator
2. Account
3. Graphql
4. User Prefs.

**Comment by theBeginner86** [comment]
> This PR is not complete yet. I've just fixed one of the two issues I identified. Will explain more in today's dev call.

**Comment by leecalcote** [comment]
> > Curious: what is `type` is used for?
> 
> `type` means the type of remote extension. Used to determine what `type` of extension to render onto Meshery UI. Could be:
> 
> 1. Navigator
> 2. Account
> 3. Graphql
> 4. User Prefs.

Aha! But, of course. Thank you.

Is this PR still `do-not-merge`?

File: ui/pages/extension/[...component].js [hunk 1]
```diff
@@ -65,69 +67,108 @@ class RemoteExtension extends React.Component {
...
+  componentDidUpdate(prevProps, prevState) {
+    if (this.props.extensionType !== prevProps.extensionType || this.props.router.query.component != prevProps.router.query.component) {
       this.renderExtension();
     }
   }
 
   renderExtension = () => {
-    getFullPageExtensions(extNames => {
+    let cap = this.props.capabilitiesRegistry;
+    // For no restricted, we will show extensions
```
Comment (theBeginner86) [inline ui/pages/extension/[...component].js:113]:
```suggestion
    // For unrestricted access, show extensions
```

**Comment by theBeginner86** [comment]
Ready

<details>
<summary>demo</summary>
<br>


https://user-images.githubusercontent.com/73700530/221369545-a2fda520-8ee6-4727-9304-81fee0c03a18.mov



</details>

File: ui/pages/extension/[...component].js [hunk 2]
```diff
@@ -65,69 +67,108 @@ class RemoteExtension extends React.Component {
...
       this.renderExtension();
     }
   }
 
   renderExtension = () => {
-    getFullPageExtensions(extNames => {
+    let cap = this.props.capabilitiesRegistry;
+    // For unrestricted access, show extensions
+    if (cap !== null && !cap?.restrictedAccess?.isMesheryUiRestricted) {
+      console.log("inside if")
```
Comment (theBeginner86) [inline ui/pages/extension/[...component].js:115]:
```suggestion

```

**Pull request merged**